lms/djangoapps/instructor_analytics/tests/test_basic.py

```diff
@@ -92,10 +93,13 @@ def test_get_response_state_with_ora(self, files_descriptions, saved_response):
             'saved_files_descriptions': json.dumps([files_descriptions]),
         })
         response = Mock(module_type='openassessment', student=Mock(username='staff'), state=payload_state)
-
         transformed_state = json.loads(get_response_state(response))
-        self.assertEqual(transformed_state['saved_files_descriptions'][0], files_descriptions)
-        self.assertEqual(transformed_state['saved_response']['parts'][0]['text'], saved_response)
+        if six.PY2:
+            self.assertEqual(transformed_state['saved_files_descriptions'][0], files_descriptions)
+            self.assertEqual(transformed_state['saved_response']['parts'][0]['text'], saved_response)
+        else:
+            self.assertEqual(json.loads(transformed_state['saved_files_descriptions'])[0], files_descriptions)
```

Comment:
in py3 json loads does not load the child instances like py2. It loads only one parent.

Reply:
See Line#96 ; 

`transformed_state` is already a loaded object why you need to load it again? 

if its a py3 difference, what is the type of `transformed_state` on line `101`?

Reply:
@awaisdar001 at that point that object is not loaded and calling transformed_state['saved_files_descriptions'][0] returns '['

Reply:
@Ayub-Khan It looks like this works:
```
>>> test = """{"a": { "b" : "c"}}"""
>>> import json
>>> json.loads(test)
{'a': {'b': 'c'}}
```
Is the data different from what I'm thinking here?  Can you provide an example.

Reply:
I think the right fix should be in [get_response_state](https://github.com/edx/edx-platform/blob/b1cdf6a96fa86c7b38255b42466b7cea81e6ee0c/lms/djangoapps/instructor_analytics/basic.py#L500) method. 
`json.dumps` no longer support `encoding` keyword argument in python3. This test would run just fine after that change.

Reply:
@awaisdar001 thanks.